src/widgetastic/widget/table.py

```diff
@@ -0,0 +1,1055 @@
+
+        Default implementation just writes a debug message that it is not used.
+        """
+        self.logger.debug('Row saving not used.')
+
+    @property
+    def has_rowcolspan(self):
+        """Checks whether table has rowspan/colspan attributes"""
+        return bool(self.browser.elements('.//td[@rowspan or @colspan]', parent=self))
+
+    def _process_table(self):
+        queue = deque()
+        tree = Node(name=self.browser.tag(self), obj=self, position=None)
+        queue.append(tree)
+
+        while len(queue) > 0:
+            node = queue.popleft()
+            # visit node
+            children = self.browser.elements('./*[descendant-or-self::node()]', parent=node.obj)
+            for position, child in enumerate(children):
+                cur_tag = self.browser.tag(child)
+                if cur_tag == 'tr':
+                    # todo: add logger
+                    # position has been decremented for rows because it is incremented
+                    # if 0 in TableRow.__init__ for some reason
+                    cur_obj = TableRow(parent=self._get_ancestor_node_obj(node), index=position)
+                    cur_node = Node(name=cur_tag, parent=node, obj=cur_obj, position=position)
+                    queue.append(cur_node)
+                elif cur_tag == 'td':
+                    cur_position = self._get_position_respecting_spans(node)
+                    cur_obj = TableColumn(parent=node.obj, position=cur_position)
+                    Node(name=cur_tag, parent=node, obj=cur_obj, position=cur_position)
+
+                    rowsteps = range(1, int(child.get_attribute('rowspan') or 0))
+                    colsteps = range(1, int(child.get_attribute('colspan') or 0))
+                    coordinates = set(six.moves.zip_longest(colsteps, rowsteps, fillvalue=0))
+
+                    # when there are both rowspan and colspan set, we need to generate additional
+                    # cell references
+                    additional_coordinates = set()
+                    for col, row in coordinates:
+                        if col > 0 and row > 0:
+                            for new_coord in itertools.product(range(col+1), range(row+1)):
+                                if new_coord == (0, 0):
+                                    continue
+                                additional_coordinates.add(new_coord)
+
+                    coordinates = coordinates | additional_coordinates
+
+                    for col_step, row_step in coordinates:
+                        if row_step >= 1:
+                            ref_parent = self._get_sibling_node(node, steps=row_step)
+                        else:
+                            ref_parent = node
+                        ref_obj = TableReference(parent=ref_parent, reference=cur_obj)
+                        ref_position = cur_position if col_step == 0 else cur_position + col_step
+                        Node(name='ref', parent=ref_parent, obj=ref_obj,
```

Comment:
Again, not sure why this object is instantiated but not assigned.

Reply:
it isn't necessary since all interaction is done thru root node